plugins/cloudtrail/cloudtrailEnabled.js

```diff
@@ -2,77 +2,87 @@ var async = require('async');
-					var trail = describeTrails.data[t];
-
-					var getTrailStatus = helpers.addSource(cache, source,
-					    ['cloudtrail', 'getTrailStatus', region, trail.TrailARN]);
-
-					if (getTrailStatus && getTrailStatus.data &&
-						getTrailStatus.data.IsLogging) {
-						helpers.addResult(results, 0, 'CloudTrail is enabled', region);
-
-						if (trail.IncludeGlobalServiceEvents) {
-							globalServicesMonitored = true;
-						}
-
-						found = true;
-						break;
-					}
-				}
-
-				if (!found) {
-					helpers.addResult(results, 2, 'CloudTrail is setup but is not logging API calls', region);
-				}
-			}
-			
-			rcb();
-		}, function(){
-			if (!globalServicesMonitored) {
-				helpers.addResult(results, 2, 'CloudTrail is not monitoring global services');
-			} else {
-				helpers.addResult(results, 0, 'CloudTrail is monitoring global services');
-			}
-
-			callback(null, results, source);
-		});
-	}
+        'containing sensitive data. CloudTrail is the recommended ' +
+        'logging and auditing solution for AWS since it is tightly ' +
+        'integrated into most AWS services and APIs.'
+    },
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var regions = helpers.regions(settings.govcloud);
+
+        var globalServicesMonitored = false;
+
+        async.each(regions.cloudtrail, function(region, rcb){
+            var describeTrails = helpers.addSource(cache, source,
+                ['cloudtrail', 'describeTrails', region]);
+
+            if (!describeTrails) return rcb();
+
+            if (describeTrails.err || !describeTrails.data) {
+                helpers.addResult(results, 3,
+                    'Unable to query for CloudTrail policy: ' + helpers.addError(describeTrails), region);
+                return rcb();
+            }
+
+            if (!describeTrails.data.length) {
+                helpers.addResult(results, 2, 'CloudTrail is not enabled', region);
+            } else {
+                // Ensure logging is enabled
+                var found;
+                var globalEnabled = false;
+
+                for (t in describeTrails.data) {
+                    var trail = describeTrails.data[t];
+
+                    if (trail.IncludeGlobalServiceEvents) {
+                        globalServicesMonitored = true;
+                        globalEnabled = true;
+                    }
+
+                    var getTrailStatus = helpers.addSource(cache, source,
+                        ['cloudtrail', 'getTrailStatus', region, trail.TrailARN]);
+
+                    if (getTrailStatus && getTrailStatus.data &&
+                        getTrailStatus.data.IsLogging) {
+
+                        helpers.addResult(results, 0, 'CloudTrail is enabled', region);
+
+                        if (trail.IncludeGlobalServiceEvents) {
+                            helpers.addResult(results, 0, 'CloudTrail is enabled to monitor global services', region);
```

Comment:
Wouldn't this duplicate results?

If `IncludeGlobalServiceEvents` is true, then a result gets added here and then again in line 82.